config_converter/config_converter.py

```diff
@@ -0,0 +1,213 @@
+        self.main_map = {
+            'in_tail': {
+                'exclude_path': 'exclude_path',
+                'path': 'path',
+                'path_key': 'path_field_name',
+                'pos_file': 'checkpoint_file',
+                'refresh_interval': 'refresh_interval',
+                'rotate_wait': 'rotate_wait'
+            }
+        }
+        # each sub list contains fields that need to be treated differently
+        self.special_map = {
+            'in_tail': [
+                'format', 'format_firstline', '@type',
+                'multiline_flush_interval', 'expression'
+            ] + [f'format{i}' for i in range(1, 21)]
+        }
+        # dictionary of dictionaries - one per main plugin type (source, match)
+        # each dict contains fields that go in the upper level
+        self.outer_map = {'source': {'tag': 'name', '@type': 'type'}}
+        # each sub list contains known and allowed sub directives
+        self.supported_dirs = {'in_tail': ['parse']}
+        self.unsupported_fields = [
+            'emit_unmatched_lines', 'enable_stat_watcher',
+            'enable_watch_timer', 'encoding', 'from_encoding',
+            'ignore_repeated_permission_error', 'limit_recently_modified',
+            'open_on_every_update', 'path_timezone',
+            'pos_file_compaction_interval', 'read_from_head',
+            'read_lines_limit', 'skip_refresh_on_startup'
+        ]
+        self.supported_plugins = ['in_tail']
+        self.unsupported_plugins = ['in_forward', 'in_syslog']
+
+    def parser_dir(self, specific: dict, p: config_pb2.Param) -> None:
+        """Create parser dir in master config."""
+        parser_type_map = {
+            'multiline': 'multiline',
+            'none': 'none',
+            'regex': 'regex',
+            'apache2': 'regex',
+            'apache_error': 'regex',
+            'json': 'json',
+            'nginx': 'regex'
+        }
+        specific['parser'] = specific.get('parser', {})
+        if p.name == 'expression':
+            specific['parser']['regex_parser_config'] = \
+                    specific['parser'].get('regex_parser_config', {})
+            specific['parser']['regex_parser_config'][p.name] = p.value
+        elif p.name == 'format' or p.name == '@type':
+            if p.value not in parser_type_map:
+                print('unknown parser format type ' + p.value)
+            else:
+                specific['parser']['type'] = parser_type_map[p.value]
+        else:
+            specific['parser']['multiline_parser_config'] = \
+                    specific['parser'].get('multiline_parser_config', {})
+            if p.name[-1].isdigit():
+                specific['parser']['multiline_parser_config']\
+                        ['format_' + p.name[6:]] = p.value
+            elif p.name == 'multiline_flush_interval':
+                specific['parser']['multiline_parser_config'][p.name[10:]] = \
```

Comment:
```suggestion
                specific['parser']['multiline_parser_config']['flush_internal'] = \
```

I think it would be more clear.

Reply:
Agreed - will change that!